tests/integration/test_stream.py

```diff
@@ -0,0 +1,59 @@
+#
```

Comment:
We deleted `test_concurrent` because it had no purpose other than a debugging playground, and we already have `test_debug` for that. Now we introduce `test_stream`, with the goal of one day running it in CI as it contains a useful test case (that failed without this PR, but passes with this PR). However we're not adding it to CI just yet as there is substantial technical debt in the test infrastructure blocking it and we have other priorities.